File: test-files/golden-tests/generator/hbs/lua-helper/addons/lua/generator/common/helpers/glue.lua
-- Glue helper: joins positional args using the first argument as separator.
-- Uses normalize_args from _utils.lua.

return function(...)
    local list = normalize_args(...)
    if list.n == 0 then
        return ""
    end
    local sep = tostring(list[1])
    local items = {}
    for i = 2, list.n do
        items[#items + 1] = tostring(list[i])
    end
    return table.concat(items, sep)
end

File: tests/golden/fixtures/generator/hbs/js-helper/addons/js/generator/common/helpers/glue.js
// Glue helper: flattens positional args (arrays allowed) and joins with the first argument as separator.
// Uses normalize_args from _utils.js (loaded before helper files).

function glue()
{
    var list = normalize_args(arguments);
    if (list.length === 0)
        return '';

    var sep = list[0];
    var items = [];
    for (var i = 1; i < list.length; ++i)
    {
        var v = list[i];
        if (Array.isArray(v))
        {
            for (var j = 0; j < v.length; ++j)
                items.push(v[j]);
        }
        else
        {
            items.push(v);
        }
    }
    return items.join(sep);
}
